### PR TITLE
feat: upgrade to sql-formatter v15, add `dataTypeCase` and `keywordCase` options

### LINKED ISSUE #326: prettier-plugin-sql: Upgrade to sql-formatter@15
There is another new version [`sql-formatter@15.0.0`](https://github.com/sql-formatter-org/sql-formatter/releases/tag/v15.0.0), which introduces [new `dataTypeCase` and `functionCase` options](https://github.com/sql-formatter-org/sql-formatter/pull/673)

Some breaking changes:

> Breaking changes
>
> - The keywordCase option no more applies to data types.
> - The identifierCase option no more applies to function names.
> - New formatting style of CREATE TABLE (https://github.com/sql-formatter-org/sql-formatter/issues/495)
> - New formatting style of CREATE VIEW
> - When using custom dialects: TokenizerOptions has new mandatory reservedDataType field.

I'd be happy to do a PR with these changes, after the PR documenting SQL-in-JS/TS with `prettier-plugin-embed`

### PR DESCRIPTION
**Pull request by karlhorky**
Closes #326

### CODE REVIEW

**Comment by JounQin** [comment]
@karlhorky Please fix the CI issue about formatting.

File: packages/sql/test/__snapshots__/fixtures.spec.ts.snap [hunk 1]
```diff
@@ -1,16 +1,14 @@
...
+"CREATE TABLE \\"test\\" (\\"id\\" uuid NOT NULL)
 WITH
   (oids = false);
 "
 `;
 
 exports[`parser and printer > should format all fixtures > 233.sql 1`] = `
-"CREATE TABLE
-  \\"test\\" (\\"id\\" uuid NOT NULL)
+"CREATE TABLE \\"test\\" (\\"id\\" uuid NOT NULL)
```
Comment (karlhorky) [inline packages/sql/test/__snapshots__/fixtures.spec.ts.snap:11]: these are the new format of `CREATE TABLE` mentioned in [the `sql-formatter@15.0.0` release notes](https://github.com/sql-formatter-org/sql-formatter/releases/tag/v15.0.0):

> New formatting style of CREATE TABLE (sql-formatter-org/sql-formatter#495) Thanks to Timon Jurschitsch

**Comment by karlhorky** [comment]
@JounQin I think that this is ready to go.

Before the release, maybe `prettier-plugin-sql@0.18.0` should also include [the `dialect` option](https://github.com/un-ts/prettier/issues/322)?

**Comment by JounQin** [comment]
@karlhorky If you have time to work on that new feature, I'll wait for it before releasing.

**Comment by karlhorky** [comment]
@JounQin ah, I mentioned in #322 that it would be great if you could take over the `dialect` option, not sure when I'll get to it

**Comment by JounQin** [comment]
OK, I'll take a look when I'm free.

**Comment by karlhorky** [comment]
I can confirm this is working in [`prettier-plugin-sql@0.18.0`](https://github.com/un-ts/prettier/releases/tag/prettier-plugin-sql%400.18.0) 🎉 Thanks for the release.

**Pull request merged**